fix bug of model `infloat/e5-mistral-7b-instruct` return error 'MistralConfig' object has no attribute 'rope_scaling'`'

kaixuanliu opened the pull request:
fix the bug that when loading model `infloat/e5-mistral-7b-instruct`, optimum habana returns error like
`File "/usr/src/backends/python/server/text_embeddings_server/models/__init__.py", line 69, in get_model
    return DefaultModel(model_path, device, dtype)

  File "/usr/src/backends/python/server/text_embeddings_server/models/default_model.py", line 24, in __init__
    model = AutoModel.from_pretrained(model_path).to(dtype).to(device)

  File "/usr/local/lib/python3.10/dist-packages/transformers/models/auto/auto_factory.py", line 563, in from_pretrained
    return model_class.from_pretrained(

  File "/usr/local/lib/python3.10/dist-packages/transformers/modeling_utils.py", line 3550, in from_pretrained
    model = cls(config, *model_args, **model_kwargs)

  File "/usr/local/lib/python3.10/dist-packages/transformers/models/mistral/modeling_mistral.py", line 916, in __init__
    [MistralDecoderLayer(config, layer_idx) for layer_idx in range(config.num_hidden_layers)]
                                                                                                                                                                                                                                                                                              File "/usr/local/lib/python3.10/dist-packages/transformers/models/mistral/modeling_mistral.py", line 916, in <listcomp>
    [MistralDecoderLayer(config, layer_idx) for layer_idx in range(config.num_hidden_layers)]

  File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/mistral/modeling_mistral.py", line 470, in __init__
    self.self_attn = GaudiMistralAttention(config, layer_idx)                                                                                                                                                                                                                               
  File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/mistral/modeling_mistral.py", line 232, in __init__
    self._init_rope()

  File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/mistral/modeling_mistral.py", line 243, in _init_rope
    if self.config.rope_scaling is None:

  File "/usr/local/lib/python3.10/dist-packages/transformers/configuration_utils.py", line 263, in __getattribute__
    return super().__getattribute__(key)

AttributeError: 'MistralConfig' object has no attribute 'rope_scaling'`

kaixuanliu commented:
@regisss , pls help review

regisss reviewed (commented):
We would probably need to add a call to `adapt_transformers_to_gaudi` in TEI. But it looks good to me.
Can you also run `make style` please?

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/optimum-habana/pr_1042). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

kaixuanliu commented:
> We would probably need to add a call to `adapt_transformers_to_gaudi` in TEI. But it looks good to me. Can you also run `make style` please?

Have run `make style` and fixed the style issue. On the TEI side, we have added `adapt_transformers_to_gaudi` [here](https://github.com/huggingface/tei-gaudi/blob/habana-main/backends/python/server/text_embeddings_server/models/default_model.py#L22), but we still meet this error without this PR.

kaixuanliu commented:
close this PR as [here](https://github.com/huggingface/optimum-habana/commit/93f1fcf934d7b3b24cf0a0b9cac8e232af6ccc27) has solved this bug
